client/app/bundles/course/assessment/submission/reducers/liveFeedback.js

```diff
@@ -6,120 +6,138 @@ const initialState = {
+    case actions.LIVE_FEEDBACK_USER_REQUEST: {
+      const { questionId, answerId, userRequest } = action.payload;
       return produce(state, (draft) => {
-        draft.feedbackByQuestion[questionId] = {
-          ...draft.feedbackByQuestion[questionId],
-          isRequestingLiveFeedback: false,
+        const updatedConversation = [
+          ...getConversation(draft, questionId),
+          {
+            text: userRequest,
+            sender: 'Student',
+            timestamp: new Date().toISOString(),
+            index: getConversation(draft, questionId).length,
+          },
+        ];
+        updateFeedbackForQuestion(draft, questionId, {
+          isRequestingLiveFeedback: true,
           pendingFeedbackToken: null,
-        };
-      });
-    }
-    case actions.LIVE_FEEDBACK_ITEM_MARK_RESOLVED: {
-      const { questionId, lineId, path } = action.payload;
-      return produce(state, (draft) => {
-        if (path in draft.feedbackByQuestion[questionId].feedbackFiles) {
-          draft.feedbackByQuestion[questionId].feedbackFiles[path] =
-            draft.feedbackByQuestion[questionId].feedbackFiles[path].map(
-              (line) =>
-                line.id === lineId ? { ...line, state: 'resolved' } : line,
-            );
-        }
-      });
-    }
-    case actions.LIVE_FEEDBACK_ITEM_MARK_DISMISSED: {
-      const { questionId, lineId, path } = action.payload;
-      return produce(state, (draft) => {
-        if (path in draft.feedbackByQuestion[questionId].feedbackFiles) {
-          draft.feedbackByQuestion[questionId].feedbackFiles[path] =
-            draft.feedbackByQuestion[questionId].feedbackFiles[path].map(
-              (line) =>
-                line.id === lineId ? { ...line, state: 'dismissed' } : line,
-            );
-        }
+          answerId,
+          conversation: updatedConversation,
+          suggestedReplies: [],
+          isShowingPopup: true,
+        });
       });
     }
-    case actions.LIVE_FEEDBACK_ITEM_DELETE: {
-      const { questionId, lineId, path } = action.payload;
+    case actions.LIVE_FEEDBACK_OPEN_POPUP: {
+      const { questionId, isShowingPopup } = action.payload;
       return produce(state, (draft) => {
-        if (path in draft.feedbackByQuestion[questionId].feedbackFiles) {
-          draft.feedbackByQuestion[questionId].feedbackFiles[path] =
-            draft.feedbackByQuestion[questionId].feedbackFiles[path].filter(
-              (line) => line.id !== lineId,
-            );
-          if (
-            !draft.feedbackByQuestion[questionId].feedbackFiles[path] ||
-            draft.feedbackByQuestion[questionId].feedbackFiles[path].length ===
-              0
-          ) {
-            delete draft.feedbackByQuestion[questionId].feedbackFiles[path];
-          }
-        }
+        updateFeedbackForQuestion(draft, questionId, {
+          isShowingPopup,
+          suggestedReplies: ['I am stuck', "My code doesn't work"],
```

Comment:
Don't repeat the first two elements in the `suggestions` array, you should be only hardcoding strings like this once.

Reply:
Actually, they are only showed when first time popup open. Then after that, they are randomly generated.